src/Image.cc

```diff
@@ -604,17 +609,14 @@ Image::loadGIFFromBuffer(uint8_t *buf, unsigned len) {
     return CAIRO_STATUS_READ_ERROR;
   }
 
-  width = naturalWidth = gif->SWidth;
-  height = naturalHeight = gif->SHeight;
-
-  /* Cairo limit:
-   * https://lists.cairographics.org/archives/cairo/2010-December/021422.html
-   */
-  if (width > 32767 || height > 32767) {
+  if (static_cast<size_t>(gif->SWidth) > canvas_max_side || static_cast<size_t>(gif->SHeight) > canvas_max_side) {
```

Comment:
If you're okay with implicit conversions, you don't need the `static_cast`s here